scripts/python/qa-scripts/createVCAAggregate.py

```diff
@@ -0,0 +1,523 @@
+            if not set([self.opt.notValidCol]).issubset(data.columns):
+                data[self.opt.notValidCol] = data[self.opt.notValidAlternativeCol]
+                data.drop(self.opt.notValidAlternativeCol, axis=1, inplace=True)
+            data = self.filterVCAConficts(data, vcaFile)
+            if data is not False:
+                self.vcasData.append(data)
+                self.vcasFileList.append(vcaFile)
+
+    def createDoc(self):
+        self.loadVCAsFiles()
+        # Loop over master ids as reference
+        for id, row in self.dfVca.iterrows():
+            proposerAss = self.dfMasterProposers.loc[id]
+            # Loop over all vca files
+            for filesIdx, vcaDf in enumerate(self.vcasData):
+                if (id in vcaDf.index):
+                    locAss = vcaDf.loc[id]
+                    integrity = self.checkIntegrity(id, row, locAss)
+                    vca_fn = self.vcasFileList[filesIdx]
+                    vca_filename = os.path.basename(vca_fn.replace('\\',os.sep))
+                    single_vca = next((item for item in self.vcas if item['vca_file'] == vca_filename), None)
+                    if (integrity is False):
+                        print("{} failed to pass the integrity test at id {}".format(vca_fn, id))
+                    if integrity:
+                        bad = self.badFeedback(locAss)
+                        good = self.goodFeedback(locAss)
+                        excellent = self.excellentFeedback(locAss)
+                        if (self.isVCAfeedbackValid(locAss, bad, good, excellent)):
+                            if (bad or good or excellent):
+                                reviews_num_col = "No. of Reviews"
+                                if locAss[self.opt.challengeCol] in self.dChallenges:
+                                    reviews_num_col = reviews_num_col + " " + locAss[self.opt.challengeCol]
+                                if reviews_num_col in single_vca:
+                                    single_vca[reviews_num_col] = single_vca[reviews_num_col] + 1
+                                else:
+                                    single_vca[reviews_num_col] = 1
+                                self.dfVca.loc[id, self.opt.noVCAReviewsCol] = self.dfVca.loc[id, self.opt.noVCAReviewsCol] + 1
+                                # Append the single review to the vcaMerged file
+                                toBeMergedAssessment = locAss.copy()
+                                toBeMergedAssessment['id'] = id
+                                toBeMergedAssessment['name'] = single_vca['name']
+                                self.vcaMerged = self.vcaMerged.append(toBeMergedAssessment)
+
+                            for col in self.allColumns:
+                                colVal = self.checkIfMarked(locAss, col)
+                                if (colVal > 0):
+                                    self.dfVca.loc[id, col] = self.dfVca.loc[id, col] + colVal
```

Comment:
what's this for?

Reply:
This is the key part of the aggregation: for each vca criteria (excellent, good, filtered out) of each assessment the counter is increased if the vCA marked it.
Using `allColumns` allows to criteria the columns without changing the logic (in previous iterations the criteria were different)